network/p2p/middleware.go

```diff
@@ -347,6 +351,32 @@ func (m *Middleware) SendDirect(msg *message.Message, targetID flow.Identifier)
 		return fmt.Errorf("failed to create stream for %s: %w", targetID, err)
 	}
 
+	success := false
+
+	defer func() {
+		if success {
+			// close the stream immediately
+			err = stream.Close()
+			if err != nil {
+				err = fmt.Errorf("failed to close the stream for %s: %w", targetID, err)
+			}
+
+			// OneToOne communication metrics are reported with topic OneToOne
+			m.metrics.NetworkMessageSent(msg.Size(), metrics.ChannelOneToOne, msg.Type)
```

Comment:
Why do we remove this logic?
```
	if _, isStaked := m.ov.Identities().ByNodeID(targetID); !isStaked {
		channel = metrics.ChannelOneToOneUnstaked
	}
```

Reply:
This was from previously, when we were using same libp2p host on Access Node for both staked and public network.

Now we use two separate ones, so we don't need this anymore, because the metric name already tells us whether it is staked or public network